packages/components/aside/src/aside.js

```diff
@@ -69,5 +67,29 @@ customElementDefineHelper(
 				<ts-cover class="ts-aside-cover" ?data-visible=${this.visible} @click="${this.close}"></ts-cover>
 			`;
 		}
+
+		update(changedProperties) {
+			super.update(changedProperties);
+			if (changedProperties.has('visible')) {
+				const oldVal = changedProperties.get('visible');
+				if (oldVal === false) {
```

Comment:
Can you refactor this to something like this?
const eventName = oldVal ? customEventNames.CLOSE :  customEventNames.OPEN;
this.dispatchCustomEvent(eventName)

Reply:
This is not if else. The first time it is undefined. I can't trigger Close .